src/sql/services/common/commonServiceInterface.service.ts

```diff
@@ -0,0 +1,279 @@
+	protected _modelViewService = this._bootstrapService.modelViewService;
+	protected _partService = this._bootstrapService.partService;
+	protected _angularEventingService = this._bootstrapService.angularEventingService;
+	protected _environmentService = this._bootstrapService.environmentService;
+
+	/* Special Services */
+	protected _metadataService: SingleConnectionMetadataService;
+	protected _connectionManagementService: SingleConnectionManagementService;
+	protected _adminService: SingleAdminService;
+	protected _queryManagementService: SingleQueryManagementService;
+	protected _contextKeyService: IContextKeyService;
+
+	protected _connectionContextKey: ConnectionContextkey;
+
+	constructor(
+		@Inject(BOOTSTRAP_SERVICE_ID) protected _bootstrapService: IBootstrapService
+	) {
+		super();
+	}
+
+	public get notificationService(): INotificationService {
+		return this._notificationService;
+	}
+
+	public get configurationEditingService(): ConfigurationEditingService {
+		return this._configurationEditingService;
+	}
+
+	public get metadataService(): SingleConnectionMetadataService {
+		return this._metadataService;
+	}
+
+	public get connectionManagementService(): SingleConnectionManagementService {
+		return this._connectionManagementService;
+	}
+
+	public get commandService(): ICommandService {
+		return this._commandService;
+	}
+
+	public get themeService(): IWorkbenchThemeService {
+		return this._themeService;
+	}
+
+	public get contextMenuService(): IContextMenuService {
+		return this._contextMenuService;
+	}
+
+	public get instantiationService(): IInstantiationService {
+		return this._instantiationService;
+	}
+
+	public get modelViewService(): IModelViewService {
+		return this._modelViewService;
+	}
+
+	public get partService(): IPartService {
+		return this._partService;
+	}
+
+	public get contextKeyService(): IContextKeyService {
+		return this._contextKeyService;
+	}
+
+	public get adminService(): SingleAdminService {
+		return this._adminService;
+	}
+
+	public get queryManagementService(): SingleQueryManagementService {
+		return this._queryManagementService;
+	}
+
+	public get environmentService(): IEnvironmentService {
+		return this._environmentService;
+	}
+
+	public get contextViewService(): IContextViewService {
+		return this._contextViewService;
+	}
+
+	public get workspaceContextService(): IWorkspaceContextService {
+		return this._workspaceContextService;
+	}
+
+	public get storageService(): IStorageService {
+		return this._storageService;
+	}
+
+	public get capabilitiesService(): ICapabilitiesService {
+		return this._capabilitiesService;
+	}
+
+	public get angularEventingService(): IAngularEventingService {
+		return this._angularEventingService;
+	}
+
+	/**
+	 * Set the selector for this dashboard instance, should only be set once
+	 */
+	public set selector(selector: string) {
+		this._uniqueSelector = selector;
+		this._getbootstrapParams();
+	}
+
+	protected _getbootstrapParams(): void {
```

Comment:
Is this different in any way from the sub-class implementation in DashboardServiceInterface? If not can we remove the subclass implementation? Or at a minimum, in the subclass call the base function here for common setup, then do the specialized calls in the overriding method?

Reply:
yes the implementation is different for dashboard 

---
In reply to: [182227147](https://github.com/Microsoft/sqlopsstudio/pull/1168#discussion_r182227147) [](ancestors = 182227147)